x/qgb/orchestrator/querier.go

```diff
@@ -221,6 +229,11 @@ func (q *querier) QueryLastValset(ctx context.Context) (types.Valset, error) {
 		return types.Valset{}, err
 	}
 
+	if len(lastValsetResp.Valsets) == 1 && lastValsetResp.Valsets[0].Nonce == 1 {
```

Comment:
Is there a case where `len(lastValsetResp.Valsets) == 1 && lastValsetResp.Valsets[0].Nonce != 1` is `true`?

Reply:
Yes, when we just start the chain and only have one valset.

Reply:
In that case `lastValsetResp.Valsets[0].Nonce` would be `1`?

Reply:
yes.
This case is for the following scenario:
- You start a Celestia chain. It creates on block 0 `valset.Nonce=1`
- You deploy a QGB contract using that chain
- The QGB contract will now have `valset.Nonce=1` as `valset0` 
- Then, when you want to relay `valset.Nonce=1` to the QGB
- To do so, you need to get the signatures of `valset1` using validators from `valset1` (which is `valset0` in the QGB contract)

Maybe we can handle this case differently somewhere else. Or we can update the QGB contract to have `valset.Nonce = 1` as the init valset.

Live example on Rinkeby: https://rinkeby.etherscan.io/address/0xEAcC249c0831f8DE0B6afe27e6179Ca29Dec9b74#events
Check the first two events:
- First event is the deployment event. It has nonce 0
- Second event is relaying `valset.Nonce=1`  
You can see that the threshold and the validator set hash are similar.

Now I think of it, this probably needs to be documented somewhere. Let me know if we want to document this, or we want to update the QGB contract to start at valset 1.

Reply:
Aaah now I see it. You're right, it is redundant. It can be as follows
```suggestion
	if len(lastValsetResp.Valsets) == 1 {
```

Reply:
Also, I created the following issue to track this: https://github.com/celestiaorg/celestia-app/issues/373